app/uk/gov/hmrc/economiccrimelevyregistration/controllers/EntityTypeController.scala

```diff
@@ -64,14 +64,16 @@ class EntityTypeController @Inject() (
       .fold(
         formWithErrors => Future.successful(BadRequest(view(formWithErrors, mode))),
         entityType => {
+          val previousEntityType = request.registration.entityType
+
           val event = EntityTypeSelectedEvent(
             request.internalId,
             entityType
           ).extendedDataEvent
 
           auditService.sendEvent(event)
```

Comment:
I know this isn't you, but this is being sent in fire and forget, it needs to be moved into the for comprehension below as the system will not surface any errors possibly until after the controller has already returned a result to the users.

Reply:
Moved.